matchms/filtering/add_retention.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any

```

Comment:
Looks good! Clear functions that were easy to follow for me, below some suggestions for solutions for weird exceptions that you can come across in common data formats.

Reply:
Thank you for the thorough review!